documents/service.go

```diff
@@ -253,7 +253,7 @@ func (s service) getVersion(ctx context.Context, documentID, version []byte) (Mo
 }
 
 func (s service) DeriveFromCoreDocument(cd coredocumentpb.CoreDocument) (Model, error) {
-	if cd.EmbeddedData == nil {
+	if cd.EmbeddedData == nil || &cd == nil {
```

Comment:
`&cd == nil` this is never nil. we dont need to check this

Reply:
it was returning nil in a test, but I can change the test so it won't check for nil documents

Reply:
nil most probably due to cd.EmbedData being nil i think